src/dftbp/dftbplus/initprogram.F90

```diff
@@ -3644,14 +3644,22 @@ subroutine initProgramVariables(this, input, env)
         call error("Electron dynamics does not work with MD")
       end if
 
-      if (this%isRangeSep) then
-        call error("Electron dynamics does not work with range separated calculations yet.")
-      end if
-
       if (.not. this%tRealHS .and. input%ctrl%elecDynInp%tBondE) then
         call error("Bond energies during electron dynamics currently requires a real hamiltonian.")
       end if
 
+      if (this%isRangeSep) then
+        if (input%ctrl%elecDynInp%spType == tdSpinTypes%triplet) then
+          call error("Triplet excitation kick currently disabled for range-separated functionals")
```

Comment:
Just 'kicks'? Other types of perturbations may be blocked as well (probably they are already blocked but just to be sure)